src/quickcheck.rs

```diff
@@ -0,0 +1,264 @@
+//! Implementations of the `quickcheck::Arbitrary` trait, which enables users to
+//! write tests like this, and have test values provided automatically:
+//!
+//! ```
+//! use quickcheck::quickcheck;
+//! use time::Date;
+//!
+//! struct DateRange {
+//!     from: Date,
+//!     to: Date,
+//! }
+//!
+//! impl DateRange {
+//!     fn new(from: Date, to: Date) -> Result<Self, ()> {
+//!         Ok(DateRange { from, to })
+//!     }
+//! }
+//!
+//! quickcheck! {
+//!     fn date_range_is_well_defined(from: Date, to: Date) -> bool {
+//!         let r = DateRange::new(from, to);
+//!         if from <= to {
+//!             r.is_ok()
+//!         } else {
+//!             r.is_err()
+//!         }
+//!     }
+//! }
+//! ```
+//!
+//! An implementation for [`Instant`] is intentionally omitted since its values
+//! are only meaningful in relation to a [`Duration`], and obtaining an
+//! [`Instant`] from a [`Duration`] is very simple anyway.
+
+use crate::{
+    date::{Date, MAX_YEAR, MIN_YEAR},
+    duration::Duration,
+    offset_date_time::OffsetDateTime,
+    primitive_date_time::PrimitiveDateTime,
+    time_mod::Time,
+    utc_offset::UtcOffset,
+    util::days_in_year,
+    weekday::Weekday,
+};
+use quickcheck::{Arbitrary, Gen};
+use rand::Rng;
+use std::{
+    cmp::{max, min},
+    convert::TryFrom,
+};
+
+impl Arbitrary for Date {
+    fn arbitrary<G>(g: &mut G) -> Self
+    where
+        G: Gen,
+    {
+        let y_size: i32 = i32::try_from(g.size()).unwrap_or(i32::max_value());
+        let y: i32 = if y_size == 0 {
```

Comment:
Could we avoid single-letter variable names? It generally makes things easier to read.

Reply:
Sure. Should I do the same for the `Gen` parameters?

Reply:
Nah, since that's a "standard" parameter, if that makes sense. Sorta similar to using `T` for a type parameter.

Reply:
Done. I left the closure parameters as is, though, as those are extremely local (usually within one line). Let me know if you'd like me to change those too.